src/factory/CirclesBackingFactory.sol

```diff
@@ -0,0 +1,365 @@
+    {
+        if (!supportedBackingAssets[backingAsset]) revert UnsupportedBackingAsset(backingAsset);
+
+        address instance = deployCirclesBacking(backer);
+
+        // transfer USDC.e
+        IERC20(USDC).transferFrom(backer, instance, TRADE_AMOUNT);
+        // transfer stable circles
+        IERC20(stableCRCAddress).transfer(instance, stableCRCAmount);
+
+        // create order
+        (, bytes32 appData) = getAppData(instance);
+        // Generate order UID using the "getUid" contract
+        (bytes32 orderDigest,) = GET_UID_CONTRACT.getUid(
+            USDC, // sellToken
+            backingAsset, // buyToken
+            instance, // receiver
+            TRADE_AMOUNT, // sellAmount
+            1, // buyAmount: Determined by off-chain logic or Cowswap solvers
+            VALID_TO, // order expiry
+            appData, // appData hash
+            0, // FeeAmount
+            true, // IsSell
+            false // PartiallyFillable
+        );
+        // Construct the order UID
+        bytes memory orderUid = abi.encodePacked(orderDigest, instance, uint32(VALID_TO));
+        // Initiate backing
+        CirclesBacking(instance).initiateBacking(
+            backer, backingAsset, stableCRCAddress, orderUid, USDC, TRADE_AMOUNT, stableCRCAmount
+        );
+        emit CirclesBackingInitiated(backer, instance, backingAsset, stableCRCAddress);
+    }
+
+    // LBP logic
+
+    /// @notice Creates LBP with underlying assets: `backingAssetAmount` backingAsset(`backingAsset`) and `CRC_AMOUNT` InflationaryCircles(`personalCRC`).
+    /// @param personalCRC Address of InflationaryCircles (stable ERC20) used as underlying asset in lbp.
+    /// @param backingAsset Address of backing asset used as underlying asset in lbp.
+    /// @param backingAssetAmount Amount of backing asset used in lbp.
+    function createLBP(address personalCRC, uint256 personalCRCAmount, address backingAsset, uint256 backingAssetAmount)
+        external
+        returns (address lbp, bytes32 poolId, IVault.JoinPoolRequest memory request, address vault)
+    {
+        address backer = backerOf[msg.sender];
+        if (backer == address(0)) revert OnlyCirclesBacking();
+
+        // prepare inputs
+        IERC20[] memory tokens = new IERC20[](2);
+        bool tokenZero = personalCRC < backingAsset;
+        tokens[0] = tokenZero ? IERC20(personalCRC) : IERC20(backingAsset);
+        tokens[1] = tokenZero ? IERC20(backingAsset) : IERC20(personalCRC);
+
+        uint256[] memory weights = new uint256[](2);
+        weights[0] = tokenZero ? WEIGHT_1 : WEIGHT_99;
+        weights[1] = tokenZero ? WEIGHT_99 : WEIGHT_1;
+
+        // create LBP
+        lbp = LBP_FACTORY.create(
+            _name(personalCRC),
+            _symbol(personalCRC),
+            tokens,
+            weights,
+            SWAP_FEE,
+            msg.sender, // lbp owner
+            true // enable swap on start
+        );
+
+        emit LBPDeployed(msg.sender, lbp);
```

Comment:
can we emit `backer` also in this event, so that it s easier to index on `backer` avatar?